cpp/demo/poisson_matrix_free/CMakeLists.txt

```diff
@@ -35,6 +36,9 @@ else()
     set(SCALAR_TYPE "--scalar_type=float32")
   endif()
 endif()
+
+set(SCALAR_TYPE "--scalar_type=float64")
```

Comment:
See above comment on generating o CMake files.

Reply:
@chrisrichardson kindly offered to help port the C++ demos including rethinking this bit.

Reply:
Will leave for another PR - not yet done on Unix.